core/shared/src/main/scala/org/http4s/Charset.scala

```diff
@@ -47,6 +47,8 @@ private[http4s] trait CharsetCompanionPlatform // bincompat shim
 
 object Charset extends CharsetCompanionPlatform {
 
+  def apply(nioCharset: NioCharset): Charset = new Charset(nioCharset)
```

Comment:
Maybe after making the constructor of the `Charset` package private, this `apply` is not needed?

Reply:
We can deprecate it I think, but if we remove it I think it could break source-compatibility if someone was using it?